### PR TITLE
Improve tinker command

### PR DESCRIPTION
**Pull request by tyler36**
## How this PR Solves The Problem:
- Add `Drupal10` support
- Allow Drush to accept arguments

Currently, there are issues with quotes and working with bash.  There are some documented workaround in this command but they are not idea.

https://github.com/drud/ddev/pull/3603 opens up the possibility of support so this is kinda premptive and offers another real-world case for testing and support.

Ideally, we should be able to run the examples on the [Drush php:eval](https://www.drush.org/latest/commands/php_eval/) page:

```shell
ddev tinker "$node = \Drupal\node\Entity\Node::load(1); print $node->getTitle();"
ddev tinker "node_access_rebuild()"
```

## Manual Testing Instructions:

```shell
gitpod /workspace/d9simple (main) $ ddev tinker "dump(4+4)"
^ 8
```

## Related Issue Link(s):

https://github.com/drud/ddev/pull/3603

### CODE REVIEW

**Comment by rfay** [comment]
Is this ready now with ddev v1.19?

**Comment by tyler36** [comment]
There are issues with `(` & `)` but i suspect that relates back to https://github.com/drud/ddev/pull/3603

```shell
$ ddev tinker node_access_rebuild()
\bash: syntax error near unexpected token `('

$ ddev tinker 'node_access_rebuild()'
bash: -c: line 1: syntax error near unexpected token `('
bash: -c: line 1: `set -eu && ( /mnt/ddev_config/.global_commands/web/tinker node_access_rebuild())'

$ ddev tinker "node_access_rebuild()"
bash: -c: line 1: syntax error near unexpected token `('
bash: -c: line 1: `set -eu && ( /mnt/ddev_config/.global_commands/web/tinker node_access_rebuild())'

$ ddev tinker '"node_access_rebuild()"'
 [notice] Message: Content permissions have been rebuilt.

$ ddev tinker "'node_access_rebuild()'"
 [notice] Message: Content permissions have been rebuilt.

$ ddev tinker 'node_access_rebuild\(\)'
 [notice] Message: Content permissions have been rebuilt.
```

There are notes under the "laravel" section with examples.

This PR is still valid though since it still does add 2 features:

- Add Drupal10 support
- Allow Drush to accept arguments

**Comment by tyler36** [comment]
yes ... it's good to go.

File: custom-commands/tinker/web/tinker [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Description: Run project PHP REPL. Arguments are accepted in Laravel projects (with caveats)
 ## Usage: tinker
 ## Example: "ddev tinker"
-## ProjectTypes: laravel,drupal7,drupal8,drupal9
+## ProjectTypes: laravel,drupal7,drupal8,drupal9,drupal10
```
Comment (rfay) [inline custom-commands/tinker/web/tinker:6]: You'll want to add ExecRaw I think - should solve your exec problems.

```suggestion
## ProjectTypes: laravel,drupal7,drupal8,drupal9,drupal10
## ExecRaw: true
```

**Review by rfay — CHANGES_REQUESTED** [review]

**Comment by tyler36** [comment]
Just been playing around with the command.

TLDR:
- Works on WSL & Win10 hosts
- Works on Drupal & Laravel projects
- Arguments should be wrapped with `'`; `"` works too, sometimes

## Window 10 Host

### Laravel project
```shell
 ❯  ddev tinker 6+5
11

 ❯  ddev tinker User::find(1)
[!] Aliasing 'User' to 'App\Models\User' for this Tinker session.

   Psy\Exception\FatalErrorException

  PHP Fatal error:  Class constant 'User::find' not found in eval()'d code on line 1

  at :1
  ➜   1▕

 ❯  ddev tinker 'User::find(1)'
[!] Aliasing 'User' to 'App\Models\User' for this Tinker session.
App\Models\User^ {#4271
...

 ❯  ddev tinker "User::find(1)"
[!] Aliasing 'User' to 'App\Models\User' for this Tinker session.
App\Models\User^ {#4271
...
```

### Drupal 9 project
```shell
 ❯  ddev tinker 'node_access_rebuild()'
 [notice] Message: Content permissions have been rebuilt.

 ❯ ddev tinker "$node = \Drupal\node\Entity\Node::load(1); print $node->getTitle();"
 [error]  ParseError: syntax error, unexpected token "=", expecting end of file in Drush\Commands\core\PhpCommands->evaluate() (line 1 of /var/www/html/vendor/drush/drush/src/Commands/core/PhpCommands.php(30) : eval()'d code) #0 [internal function]: Drush\Commands\core\PhpCommands->evaluate(' = \\Drupal\\node...', Array)

 ❯ ddev tinker '$node = \Drupal\node\Entity\Node::load(1); print $node->getTitle();'
Who Doesn’t Like a Good Waterfall?
```


## WSL Ubuntu

### Laravel project
```shell
$ ddev tinker 6+5
11

$ ddev tinker User::find(1)
-bash: syntax error near unexpected token `('

$ ddev tinker 'User::find(1)'
[!] Aliasing 'User' to 'App\Models\User' for this Tinker session.
App\Models\User^ {#4647
...

$ ddev tinker "User::find(1)"
[!] Aliasing 'User' to 'App\Models\User' for this Tinker session.
App\Models\User^ {#4647
...
```

### Drupal 9 project
```shell
$ ddev tinker 'node_access_rebuild()'
 [notice] Message: Content permissions have been rebuilt.

$ ddev tinker "node_access_rebuild()"
 [notice] Message: Content permissions have been rebuilt.

$ ddev tinker "$node = \Drupal\node\Entity\Node::load(1); print $node->getTitle();"
 [error]  ParseError: syntax error, unexpected token "=", expecting end of file in Drush\Commands\core\PhpCommands->evaluate() (line 1 of /var/www/html/vendor/drush/drush/src/Commands/core/PhpCommands.php(30)

$ ddev tinker '$node = \Drupal\node\Entity\Node::load(1); print $node->getTitle();'
Who Doesn’t Like a Good Waterfall? 
```

File: custom-commands/tinker/web/tinker (hunk 1 above)
Comment (tyler36) [reply custom-commands/tinker/web/tinker:8]: Added

Comment (rfay) [reply custom-commands/tinker/web/tinker:8]: Just so you know, you could have just clicked "commit suggestion"

**Comment by rfay** [comment]
When you do `ddev tinker "$node = \Drupal\node\Entity\Node::load(1); print $node->getTitle();"` bash is turning `\D` into a control character before ever using it...

**Comment by rfay** [comment]
If you're happy with it i'll pull it.

**Comment by tyler36** [comment]
I am happy with it.
The readme has example of usage and it is an improvement over the current version. Thank you.
